blog-posts/staticIpLambda/staticIpLambda.md

```diff
@@ -0,0 +1,193 @@
+Moreover, we will use the NodeJS [ssh2-promise](https://www.npmjs.com/package/ssh2-promise) package to perform SFTP operations in our lambda function. 💽
+
+This is what we will build 👷:
+
+![Static Lambda IP Architecture](./assets/archi.png 'Logo Kumo')
+
+Let's get to work 💪🔨!
+
+# 1️⃣ Create a VPC hosted Lambda function with a static IP
+
+## ☁️ VPC, Nat Gateway and Elastic IP
+
+All the resources will be deployed using the **Serverless Framework** and the **Typescript AWS CDK**. If you are not familiar with it, I recommend you check out this [Serverless Framework x CDK blog post](https://dev.to/kumo/serverless-framework-aws-cdk-1dnf) to learn how to conciliate the two. 🤝
+
+First, we need to create a VPC with a public subnet and a private subnet. The private subnet will host our lambda function and the public subnet will host our NAT Gateway. <br>
+
+```ts
+const vpc = new Vpc(stack, 'Vpc', {
+	vpcName: 'vpc',
+	natGateways: 1, // 👈 Automatically creates an Elastic IP
+	maxAzs: 1, // 👈 Use more if you need high availability
+	subnetConfiguration: [
+		{
+		name: 'private-subnet-1',
+		subnetType: SubnetType.PRIVATE_WITH_NAT,
+		cidrMask: 24,
+		},
+		{
+		name: 'public-subnet-1',
+		subnetType: SubnetType.PUBLIC,
+		cidrMask: 24,
+		},
+	],
+});
+```
+> 💸 **Pricing Disclaimer**
+> This architecture will cost **some money**. The Nat Gateway runs on a EC2 instance which will cost you **around 30$/month**. This price is multiplied by the number of AZs you use. 
+ 
+ I also recommend that you use CloudFormation Outputs to store the IDs of the resources you will need later on. We want to slap that *✨clean IaC code✨* in our PR 🫱.
+ 
+ ```ts
+
+export const vpcSecurityGroupOutputId = 'SgOutputId';
+export const vpcPrivateSubnetOutputId = 'VpcPvSubOutputId';
+...
+
+const privateSubnets = vpc.selectSubnets(
+	{ subnetType: SubnetType.PRIVATE_WITH_NAT }
+);
+const [privateSubnetId1] = privateSubnets.subnetIds;
+
+new CfnOutput(stack, vpcSecurityGroupOutputId, {
+	value: vpc.vpcDefaultSecurityGroup,
+});
+new CfnOutput(stack, vpcPrivateSubnet1OutputId, {
+	value: privateSubnetId1,
+});
+
+```
+
+## ⚡ Deploy the Lambda function
+
+Now that we have our VPC, we can deploy our lambda function, using the Serverless Framework. 
+
+```ts
+
+// A small util to get CF template outputs 👇
+const getOutputValue = (
+	template: CloudFormationTemplate,
+	outputId: string
+ ): string => template.Outputs?.[outputId];
+
+export const staticIpLambda = {
+  timeout: 15,
+  handler: getHandlerPath(__dirname),
+  vpc: {
+    securityGroupIds: [
+		getOutputValue(
+		 cdkAppCloudFormationTemplate,
+		 vpcSecurityGroupOutputId
+		 ),
+	],
+    subnetIds: [
+		getOutputValue(
+		  cdkAppCloudFormationTemplate,
+		  vpcPrivateSubnet1OutputId
+		  ),
+		],
+  	},
+};
+```
+
+And bam! 🎉 Our lambda function is deployed in our private subnet. <br>
```

Comment:
Does 'bam' exist in English? :sweat_smile:

Reply:
just checked it it does !